plugin/caddyfile/marshal.go

```diff
@@ -41,7 +41,13 @@ func (block *Block) write(buffer *bytes.Buffer, level int) {
 			buffer.WriteString(" ")
 		}
 
-		if strings.ContainsAny(key, ` "'`) {
+		if strings.ContainsAny(key, "\n\"") {
```

Comment:
@francislavoie 
Updated this logic, to make use of backticks whenever it can to improve readability